core/stdlib/std.ncl

```diff
@@ -1708,7 +1712,8 @@
         'Ok Dyn,
         'Error {
           message | String | optional,
-          notes | Array String | optional
+          notes | Array String | optional,
+          blame_location | Dyn | optional,
```

Comment:
One  detail: we don't have to make this optional, because in fact  `check` ensures that `blame_location` is always set. On the other hand, this would make the contract from `check` and the return value of `custom` to mismatch, which might be annoying for typed code.

Reply:
Right, the annoyance with typed code was why I left it optional, especially since I'd expect it to be common in custom contracts to just return the result of a `std.contract.check`. I think at some point we discussed subtyping for records with optional fields? If we had that, this would be nicer...